### PR TITLE
allow setting data directly into a cube cell

### PR DESCRIPTION
**Pull request by fernandoabolafio**
Adds a new method to Cube called `setSingleData` which takes an object containing a key-value pair for all dimensions. It then maps that to the correspondent index in the underlying store array. 
This is a fast operation and initially will be used for optimizing the actions operations.

### CODE REVIEW

File: src/cube.js [hunk 1]
```diff
@@ -289,6 +289,35 @@ class Cube {
         }
     }
 
+    setSingleData(measureId, coords, value) {
+        const position = this.getPosition(coords);
+
+        // validate there is a value for all dimensions
+        if (!this.dimensionIds.every(dimensionId => Boolean(coords[dimensionId]))) {
+            throw new Error(`setData: no value for all dimensions`);
+        }
```
Comment (joelazar) [inline src/cube.js]: I would first do the validation, then get the position of the coords here.

File: src/cube.js [hunk 2]
```diff
@@ -289,6 +289,35 @@ class Cube {
...
+
+        if (this.storedMeasures[measureId] === undefined) {
+            throw new Error(`setData: no such measure ${measureId}`);
```
Comment (joelazar) [inline src/cube.js]:
```suggestion
            throw new Error(`setSingleData: no such measure ${measureId}`);
```

Comment (fernandoabolafio) [reply src/cube.js]: fixed

File: src/cube.js (hunk 1 above)
Comment (fernandoabolafio) [reply src/cube.js]: fixed

**Review by joelazar — APPROVED** [review]

**Pull request merged**